frontend/src/api/djangoAPI.js

```diff
@@ -1,5 +1,59 @@
 import axios from 'axios';
 
-export default axios.create({
+/**
+ * initialize the client with the base url
+ */
+const http = axios.create({
     baseURL: process.env.REACT_APP_DJANGO_API
-})
\ No newline at end of file
+})
+
+class djangoAPI {
+
+    /**
+     * performs GET request to retrieve full videos list
+     *
+     * @returns {Promise<void>}
+     *          videos list
+     */
+    getAllVideos() {
+        return http.get('videos/');
+    }
+
+    /**
+     * performs GET request to retrieve videos list from searchbar entry
+     * 
+     * @param name 
+     *          searchbar query
+     * @returns {Promise<void>}
+     *          videos list
+     */
+    getVideosByName(name) {
+        return http.get('/videos/', {params: {search_query: name}});
+    }
+
+    /**
+     * performs GET request to retrieve a single video by it's ID
+     *
+     * @param id
+     *          video's id
+     * @returns {Promise<void>}
+     *          videos list
+     */
+    getVideosById(id) {
+        return http.get(`/videos/${id}/`);
+    }
+
+    /**
+     * performs GET request to retrieve the next videos 
+     *
+     * @param nextQuery
+     *          the full query to retrieve more videos
+     * @returns {Promise<void>}
+     *          videos list
+     */
+    getNextVideos(nextQuery) {
```

Comment:
This is hard to understand. Probably the names are not appropriate.
Are you referring to the action of getting the next page of a videos request?

This method would be better without any argument.
On way to go would be to use a Pager class, that keeps the properties needed to perform get the next or previous page.